kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ConfigurationFragment.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.util.Set;
+
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+
+/**
+ * A piece of proxy configuration which references files on the proxy container filesystem
+ * @param fragment The piece of proxy configuration
+ * @param volumes The volumes the configuration depends on
+ * @param mounts The mount the configuration depends on
+ * @param <F> The type of fragment
+ */
+public record ConfigurationFragment<F>(F fragment, Set<Volume> volumes, Set<VolumeMount> mounts) {}
```

Comment:
Is this going to be re-used in your TLS branch?

Reply:
Yes. The idea is that any node in the configuration tree _might_ reference external files (currently that's just `NameFilterDefinition`, but it will shortly also be `TargetCluster`). This allows up to construct the whole tree while keeping track of the volumes and mounts which the configuration nodes require.